miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -365,32 +333,28 @@ end
 #! Output: [offset_slot_index]
 export.apply_storage_offset
     # verify that slot_index is in bounds
-    dup.1 push.MAX_STORAGE_SLOT_INDEX lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+    dup.1 exec.memory::get_num_storage_slots lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
     # => [storage_offset, slot_index]
 
     # verify that the computed offset_slot_index is in bounds
-    add dup push.MAX_STORAGE_SLOT_INDEX lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+    add dup exec.memory::get_num_storage_slots lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
     # => [offset_slot_index]
```

Comment:
Instead of calling `memory::get_num_storage_slots` twice, I'd probably just call it once and then use `dup` to make a copy of the value.

Reply:
Duplicated and arranged stack:

```text
#! Applies offset to provided storage slot index for storage access
#!
#! Panics:
#!   - slot index is out of bounds
#!   - computed index is out of bounds
#!
#! Stack: [storage_offset, slot_index]
#! Output: [offset_slot_index]
export.apply_storage_offset
    # verify that slot_index is in bounds
    dup.1 exec.memory::get_num_storage_slots dup movdn.4 lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
    # => [storage_offset, slot_index, num_storage_slots]

    # verify that the computed offset_slot_index is in bounds
    add dup movup.2 lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
    # => [offset_slot_index]
end
```